docs/user-guide/authenticating-with-client-certificates.md

```diff
@@ -0,0 +1,82 @@
+
+You can perform authentication with client certificates.
+
+If the keyring or a truststore contains at least one valid certificate authority (CA) other than the CA of API ML, it is possible to use the client certificates issued by this CA to authenticate to API ML. This feature is not enabled by default and must be configured.
+
+:::note
+There is a limitation with respect to ACF2 systems. If you would like to offer feedback using client
+certificate authentication, please create an issue in the Zowe [api-layer repository](https://github.com/zowe/api-layer).
+:::
+
+When providing credentials during a login request with a client certificate on the same login request, the credentials take precedence and the client certificate is ignored.
+
+## How the Gateway resolves authentication 
+
+When sending a request to a service with a client certificate, the Gateway performs the following process to resolve authentication:
+
+* The client calls the service endpoint through API ML Gateway with the client certificate.
+* The client certificate and private key are checked as a valid TLS client certificate against the trusted CAs of the Gateway.
+* The public part of the provided client certificate is checked against SAF. SAF subsequently returns a user ID that owns this certificate. Z Secure Services (ZSS)* provides this API for API ML. 
+* The Gateway performs the login of the mapped user and provides valid authentication to the southbound service.
+
+:::note
+* For information about ZSS, see the section Zowe runtime in the [Zowe server-side installation overview](./install-zos).
+:::
+
+When sending a request to the login endpoint with a client certificate, the Gateway performs the following process to exchange the client certificate for an authentication token.
+
+* The client calls the API ML Gateway login endpoint with the client certificate.
+* The client certificate and private key are checked as a valid TLS client certificate against the trusted CAs of the Gateway.
+* The public part of the provided client certificate is checked against SAF. SAF subsequently returns a user ID that owns this certificate. ZSS provides this API for API ML. 
+* The Gateway performs the login of the mapped user and returns a valid JWT token.
+
+![Zowe client certificate authentication diagram](../../images/api-mediation/zowe-client-cert-auth.png)
+
+:::tip
+For more information, see the Medium blog post [Zowe client certificate authentication](https://medium.com/zowe/zowe-client-certificate-authentication-5f1c7d4d579).
+
+## Prerequisites
+
+Ensure that you satisfy the following requirements before you set up client certificate authentication:
+
+1. Specify the Zowe runtime user and set your protection by password. The user is created with the `NOPASSWORD` parameter by the Zowe installer. It is necessary to change this password. 
+
+  For RACF, issue the following TSO command:  
+
+  `ALTUSER <ZOWE_RUNTIME_USER (ZWESVUSR by default)> PASSWORD(<NEWPASSWORD>)`  
```

Comment:
Is this required for all ESMs still? @JirkaAichler @achmelo

Reply:
it is required for TSS and RACF. ACF2 will not work even with this change so it is probably irrelevant.

Reply:
Should we add a note indicating that this does not work with ACF2?